R/utilities-simulation-results.R

```diff
@@ -0,0 +1,132 @@
+}
+
+#' Get simulated values of molecules or parameters.
+#'
+#' Return values for given entities generated in a population simulation.
+#'
+#' The function receives an object of simulation results generated by running the simulation
+#' and returns time-values profiles for the chosen quantities. Results of a simulation
+#' of a single individual is treated as a population simulation with only one individual.
+#'
+#' @param simulationResults Object of type \code{SimulationResults} produced by calling \code{runSimulation}
+#' on a \code{Simulation} object.
+#' @param quantitiesOrPaths Quantity instances (element or vector) (typically retrieved using
+#' \code{getAllQuantitiesMatching}) or quantity path (element or vector of strings) for which the results are to be returned.
+#' When providing the paths, only absolute full paths are supported (i.e., no matching with '*' possible)
+#' @param individualIds \code{numeric} IDs of individiuals for which the results should be extracted.
+#' By default, all individuals from the results are considered. If the individual with the provided ID is not found, the ID is ignored
+#'
+#' @return A list of results, each entry having the name of a quantity path and consisting of the element \code{x} being the time values (in minutes)
+#' and the element \code{y} a \code{NxM} array of simulated values, with \code{N} the number of entries (equals the number of output time steps)
+#' and \code{M} the number of individuals. If no individual contains the results for a given path, the value of the element is \code{NULL}
+#'
+#' @seealso \code{\link{getResultsValues}}
+#'
+#' @export
+#'
+#' @examples
+#' simPath <- system.file("extdata", "simple.pkml", package = "ospsuite")
+#' sim <- loadSimulation(simPath)
+#' simResults <- runSimulation(sim)
+#'
+#' paths <- c(
+#'   "Organism|PeripheralVenousBlood|Caffeine|Plasma (Peripheral Venous Blood)",
+#'   "Organism|ArterialBlood|Plasma|Caffeine|Concentration in container"
+#' )
+#'
+#' resultsValues <- getPopulationResultsValues(simResults, paths, individualIds = 0)
+#' # plot(resultsValues$`Organism|PeripheralVenousBlood|Caffeine|Plasma (Peripheral Venous Blood)`)
+getPopulationResultsValues <- function(simulationResults, quantitiesOrPaths, individualIds = NULL) {
+  validateIsOfType(simulationResults, "SimulationResults")
+  quantitiesOrPaths <- c(quantitiesOrPaths)
+  validateIsOfType(quantitiesOrPaths, c("Quantity", "character"))
+  validateIsNumeric(individualIds, nullAllowed = TRUE)
+
+  # If quantities are passed, get their paths.
+  paths <- quantitiesOrPaths
+  if (isOfType(quantitiesOrPaths, "Quantity")) {
+    paths <- lapply(quantitiesOrPaths, function(x) x$path)
+  }
+  paths <- unique(paths)
+
+  # If no specific individual ids are passed, iterate through all individuals
+  if (is.null(individualIds)) {
+    nrOfIndividuals <- simulationResults$count
+    # TODO: Can we assume that all individuals ids are ascending number from 0 to the number of individuals-1??
```

Comment:
@msevestre I am really not sure if this would always work, but there was no way to get the IDs of all available individuals.

Reply:
This would not work if the simulation of a specific individual did not work. This needs to come from SimulationResults as well as  new property.